pkg/segment/structs/evaluationstructs.go

```diff
@@ -683,8 +689,50 @@ func (self *TextExpr) EvaluateText(fieldToValue map[string]utils.CValueEnclosure
 			}
 		}
 		return minString, nil
-	}
 
+	} else if self.Op == "tostring" {
+		valueStr, err := self.Val.EvaluateToString(fieldToValue)
+		if err != nil {
+			return "", fmt.Errorf("TextExpr.Evaluate: failed to evaluate value for 'tostring' operation: %v", err)
+		}
+		boolExpr, err := ParseAndEvalBooleanExpression(valueStr)
+		if err == nil {
+			return strconv.FormatBool(boolExpr), nil
+		}
```

Comment:
Do we need to evaluate boolean expressions? If so, we should probably handle that with our existing `BoolExpr` so that you don't have to reimplement boolean expressions in this `ParseAndEvalBooleanExpression()` function

Reply:
Yes made the changes